### PR TITLE
Add logical processors to custom field

### PR DESCRIPTION
**Pull request by justinbarclay**
Note that we still need to test and validate that this will run correctly in vms

### CODE REVIEW

File: unix/src/machine_stats/__init__.py [hunk 1]
```diff
@@ -222,6 +221,9 @@ def v2_runner_on_ok(self, result):
                 "storage_allocated_gb": storage_allocated_gb(facts),
                 "storage_used_gb": storage_used_gb(facts),
                 "cpu_count": cpu_count(facts),
+                "custom_fields": {
+                    "CPU_LogicalProcessors": cpu_logical_processors(facts)
```
Comment (SamDesmondKing) [inline unix/src/machine_stats/__init__.py:225]: How would you feel about making this a default server field?

**Review by SamDesmondKing — COMMENTED** [review]
Thanks for taking a look at this Justin, I think it's really important that our discovery tools have a unified approach to the data we're collecting.

I've done some [testing][(https://docs.google.com/document/d/1zLuOcnPKKOw-AubVM2jWVvbjFe5qPiCPnrqK99c-OSA/edit#heading=h.f2pa3dl125ld](https://docs.google.com/document/d/1zLuOcnPKKOw-AubVM2jWVvbjFe5qPiCPnrqK99c-OSA/edit#heading=h.55nfc2svy3i8) on this today and it seems like there are still a few kinks to work out.

Firstly, in MS4W `cpu_count` refers to physical cores, where as in MS4U it appears to refer to CPU sockets. Which one is the intention here?

Also, in my testing I found that MS4W is incorrectly counting logical cores on the `m5d.metal` instance. It's actually counting physical cores. What's strange is that for a VM `t3.2xlarge` it's correctly counting logical and physical cores. This could be a quirk of the metal instance but AWS and the machine itself seems to think it has 96 logical cores rather than the reported 48.

As I mentioned in the comments, for MS4U I've added a commit to append to the custom field key rather that overwriting it, feel free to use at your discretion.

I also have a Q about nomenclature, and whether we want to codify `cpu_logical_processors` as a first-class server attribute rather than a custom field.

File: unix/src/machine_stats/__init__.py [hunk 2]
```diff
@@ -222,6 +221,9 @@ def v2_runner_on_ok(self, result):
                 "storage_allocated_gb": storage_allocated_gb(facts),
                 "storage_used_gb": storage_used_gb(facts),
                 "cpu_count": cpu_count(facts),
+                "custom_fields": {
```
[the remark this answers is not in the document]
Comment (justinbarclay) [reply unix/src/machine_stats/__init__.py:232]: That's weird; thanks for the deep dive into this.

**Review by SamDesmondKing — APPROVED** [review]

File: windows/server_stats.ps1 [hunk 3]
```diff
@@ -167,7 +167,8 @@ $ServerStats = {
             CPU_Description        = $cpu.Description 
             CPU_Manufacturer       = $cpu.Manufacturer 
             CPU_L2CacheSize        = $cpu.L2CacheSize 
-            CPU_L3CacheSize        = $cpu.L3CacheSize 
+            CPU_L3CacheSize        = $cpu.L3CacheSize
+            CPU_LogicalProcessors  = $cpu.NumberOfLogicalProcessors
```
[the remark this answers is not in the document]
Comment (justinbarclay) [reply windows/server_stats.ps1]: Fixed

**Pull request merged**